### PR TITLE
Implement an `Image` widget

### LINKED ISSUE #48: Implement an `Image` widget
We should have a widget to simply render a `graphics::Image`:

```rust
let image = graphics::Image::new(gpu, "my_image.png");

widget::Image::new(&image).clip(Sprite { /* ... */ });
```

It should try to fill its parent and keep the correct aspect ratio at all times.

This should be similar in difficulty to #45.

### PR DESCRIPTION
**Pull request by debris**
closes #48

Screenshot 2019-08-19 at 15 45 09

### CODE REVIEW

File: src/ui/widget/image.rs [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+                height: image.height(),
+            },
+            style: Style::default().fill_width().fill_height(),
+        }
+    }
+
+    /// Sets the portion of the [`Image`] that we want to draw.
+    /// 
+    /// [`Image`]: struct.Image.html
+    pub fn clip(mut self, source: Rectangle<u16>) -> Self {
```
Comment (debris) [inline src/ui/widget/image.rs]: I'm not sure if I understood `clip` correctly, but if I did, the argument should be a `Rectangle`, not `Sprite` as it was written in #48

File: src/graphics/rectangle.rs [hunk 2]
```diff
@@ -27,4 +27,15 @@ impl Rectangle<f32> {
...
     }
+
+    /// Returns [`Point`] that is exactly in the center of this [`Rectangle`].
+    ///
+    /// [`Point`]: type.Point.html
+    /// [`Rectangle`]: struct.Rectangle.html
+    pub fn center(&self) -> Point {
+        Point::new(
+            (self.x + self.width) / 2.0, 
+            (self.y + self.height) / 2.0,
```
Comment (hecrj) [inline src/graphics/rectangle.rs]: I think this should be `self.x + self.width / 2.0`.

File: src/ui/widget/image.rs [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+    fn draw(
+        &self,
+        renderer: &mut Renderer,
+        layout: Layout<'_>,
+        _cursor_position: Point,
+    ) -> MouseCursor {
+        renderer.draw(
+            layout.bounds(),
+            self.image.clone(),
+            self.source.clone(),
```
Comment (hecrj) [inline src/ui/widget/image.rs]: `Rectangle` implements `Copy`, thus the `clone` shouldn't be necessary.

File: src/ui/renderer.rs [hunk 4]
```diff
@@ -22,6 +23,7 @@ use std::rc::Rc;
 /// [`UserInterface::configuration`]: trait.UserInterface.html#method.configuration
 pub struct Renderer {
     pub(crate) sprites: Batch,
+    pub(crate) user_sprites: Vec<Batch>,
```
Comment (hecrj) [inline src/ui/renderer.rs]: I would simply rename this to `images`.

File: src/ui/widget/image.rs (hunk 1 above)
Comment (hecrj) [reply src/ui/widget/image.rs]: Yes, you are right!

File: src/graphics/rectangle.rs (hunk 2 above)
Comment (debris) [reply src/graphics/rectangle.rs]: oh, you're absolutely right!

**Pull request merged**